File: src/main/java/wordseeker/logics/Parser.java

package wordseeker.logics;

import java.io.File;
import java.util.List;

/**
 * @author stsypanov
 */
public interface Parser {

    //override this method to enable word combination search
    public List<String> findExpression(File doc);

    //override this method to enable regexp search
    public List<String> findRegexp(File doc);


}

File: src/main/java/wordseeker/logics/Parser.kt
package wordseeker.logics

import java.io.File

/**
 * @author stsypanov
 */
interface Parser {

    //override this method to enable word combination search
    fun findExpression(doc: File): List<String>

    //override this method to enable regexp search
    fun findRegexp(doc: File): List<String>?


}
